backend/penndata/management/commands/get_engineering_events.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+import html
+import json
+
+import requests
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from penndata.models import Event
+
+
+ENGINEERING_EVENTS_WEBSITE = "https://events.seas.upenn.edu/calendar/list/"
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **kwargs):
+        try:
+            resp = requests.get(ENGINEERING_EVENTS_WEBSITE)
+        except ConnectionError:
+            print("Error:", ConnectionError)
+            return None
+
+        html_content = resp.text
+
+        start_marker = '<script type="application/ld+json">'
+        end_marker = "</script>"
+        start_index = html_content.find(start_marker)
+        end_index = html_content.find(end_marker, start_index)
+        json_ld_content = html_content[start_index + len(start_marker) : end_index]
+
+        events_data = json.loads(json_ld_content)
+
+        for event in events_data:
+            event_name = html.unescape(event.get("name", ""))
+            description = (
+                html.unescape(event.get("description", "")).replace("<p>", "").replace("</p>\n", "")
+            )
+            url = event.get("url", "")
+
+            start = datetime.datetime.fromisoformat(event.get("startDate"))
+            end = datetime.datetime.fromisoformat(event["endDate"]) if "endDate" in event else None
+
+            location = None
+            if "location" in event:
+                location = event.get("location").get("name")
+
+            email = None
```

Comment:
nit: suggest changing this to something similar to the location code by passing in an empty dict as a default